server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructConfigurationServices.test.js

```diff
@@ -0,0 +1,376 @@
+      "logical-termination-point": "ro-2-0-1-op-c-im-aa-1-0-0-003"
+    },
+    {
+      "local-id": "203",
+      "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_OUTPUT",
+      "logical-termination-point": "ro-2-0-1-op-c-im-alt-1-0-0-003"
+    },
+    {
+      "local-id": "204",
+      "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_OUTPUT",
+      "logical-termination-point": "ro-2-0-1-op-c-im-okm-1-0-0-003"
+    }
+  ]
+};
+
+const fcListArray = [{
+  uuid: 'ro-2-0-1-op-fc-bm-106',
+  name: [
+    {
+      'value-name': 'ForwardingKind',
+      value: 'core-model-1-4:FORWARDING_KIND_TYPE_INVARIANT_PROCESS_SNIPPET'
+    },
+    {
+      'value-name': 'ForwardingName',
+      value: 'PromptForBequeathingDataCausesRequestForDeregisteringOfOldRelease'
+    }
+  ],
+  'fc-port': [
+    {
+      'local-id': '000',
+      'port-direction': 'core-model-1-4:PORT_DIRECTION_TYPE_MANAGEMENT',
+      'logical-termination-point': 'ro-2-0-1-op-s-bm-001'
+    },
+    {
+      'local-id': '100',
+      'port-direction': 'core-model-1-4:PORT_DIRECTION_TYPE_INPUT',
+      'logical-termination-point': 'ro-2-0-1-op-s-im-000'
+    },
+    {
+      'local-id': '200',
+      'port-direction': 'core-model-1-4:PORT_DIRECTION_TYPE_OUTPUT',
+      'logical-termination-point': 'ro-2-0-1-op-c-bm-ro-2-0-1-002'
+    }
+  ]
+}];
+const forwardingConfigurationInputList = [
+  {
+    forwardingName: 'PromptForBequeathingDataCausesRequestForDeregisteringOfOldRelease',
+    operationClientUuid: 'ro-2-0-1-op-c-bm-ro-2-0-1-002'
+  },
+  {
+    forwardingName: 'PromptForBequeathingDataCausesRequestForBroadcastingInfoAboutServerReplacement',
+    operationClientUuid: 'ro-2-0-1-op-c-bm-ro-2-0-1-001'
+  },
+  {
+    forwardingName: 'PromptingNewReleaseForUpdatingServerCausesRequestForBroadcastingInfoAboutBackwardCompatibleUpdateOfOperation',
+    operationClientUuid: 'ro-2-0-1-op-c-bm-ro-2-0-1-003'
+  }
+];
+
+describe("configureForwardingConstructAsync", () => {
+  test("configureForwardingConstructAsync - Add Port Async", async () => {
+    const operationServerName = '/v1/embed-yourself';
+
+    jest.spyOn(OperationServerInterface, 'getOperationServerUuidAsync').mockImplementation(() => 'ro-2-0-1-op-s-is-008');
+    jest.spyOn(ForwardingDomain, 'getForwardingConstructForTheForwardingNameAsync').mockImplementation(() => fcForSubscription);
+    jest.spyOn(LogicalTerminationPoint, 'getServerLtpListAsync').mockReturnValueOnce(["ro-2-0-1-http-s-000"]);
+    jest.spyOn(HttpClientInterface, 'getApplicationNameAsync').mockReturnValueOnce('RegistryOffice');
+    jest.spyOn(HttpClientInterface, 'getReleaseNumberAsync').mockReturnValueOnce('2.0.1');
+    jest.spyOn(LogicalTerminationPoint, 'getServerLtpListAsync').mockReturnValue(["ro-2-0-1-op-c-im-ol-1-0-0-003"]);
+    jest.spyOn(HttpClientInterface, 'getApplicationNameAsync').mockReturnValue('RegistryOffice')
+    jest.spyOn(HttpClientInterface, 'getReleaseNumberAsync').mockReturnValue('2.0.2');
```

Comment:
To return different values from the same mocked method use chaining: https://jestjs.io/docs/mock-function-api#mockfnmockreturnvalueoncevalue